scripts/snap/create-snapcraft-yaml.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Ensure the script fails on errors
+set -e
+
+# Check if the correct number of arguments are passed
+if [ "$#" -ne 3 ]; then
+  echo "Usage: $0 <destination_directory> <release_tag> <artifact_path>"
+  exit 1
+fi
+
+# Assign arguments to variables
+DESTINATION_DIR="$1"
+RELEASE_TAG="$2"
+ARTIFACT_PATH="$3"
+
+# Ensure the destination directory exists
+mkdir -p "${DESTINATION_DIR}/snap"
+
+# Extract version from RELEASE_TAG, assuming it's in the format "vX.Y.Z"
+VERSION="${RELEASE_TAG#v}"
+
+# Use the provided ARTIFACT_PATH
+SOURCE="$ARTIFACT_PATH"
+
+# Create the snapcraft.yaml file
+cat > "${DESTINATION_DIR}/snap/snapcraft.yaml" <<EOF
+name: algokit-dev
+version: "$VERSION"
+summary: The AlgoKit tool for developers building on the Algorand network.
+description: |
+  AlgoKit gets developers of all levels up and running with a familiar, 
+  fun and productive development environment in minutes. 
+  The goal of AlgoKit is to help developers build and launch 
+  secure, automated production-ready applications rapidly.
+
+base: core22
+confinement: devmode
```

Comment:
```suggestion
confinement: classic
```

Reply:
Assuming we need to do this, so that we get the button in the UI that Al was mentioning

Reply:
We cannot publish with classic confinement before their approval. It will be rejected.

Reply:
@negar-abbasi actually when I tried to deploy with classic confinement, it indeed failed - but the dashboard also gave an option to start a dispute to justify classic confinement access